Source/Debug/AsyncDisplayKit+Tips.m

```diff
@@ -0,0 +1,47 @@
+//
+//  AsyncDisplayKit+Tips.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 4/12/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import "AsyncDisplayKit+Tips.h"
+#import <AsyncDisplayKit/ASDisplayNode+Ancestry.h>
+
+@implementation ASDisplayNode (Tips)
+
+static char ASDisplayNodeEnableTipsKey;
+static ASTipDisplayBlock _Nullable __tipDisplayBlock;
+
+/**
+ * Use associated objects with NSNumbers. This is a debug property - simplicity is king.
+ */
++ (void)setEnableTips:(BOOL)enableTips
+{
+  objc_setAssociatedObject(self, &ASDisplayNodeEnableTipsKey, @(enableTips), OBJC_ASSOCIATION_COPY);
+}
+
++ (BOOL)enableTips
+{
+  NSNumber *result = objc_getAssociatedObject(self, &ASDisplayNodeEnableTipsKey);
+  if (result == nil) {
+    return YES;
```

Comment:
No the default is `YES` (RTFM! 😛). The global enable is whether they define `AS_ENABLE_TIPS`. This property is only for _disabling_ tips on a per-class basis, say if we make a false-positive that they find annoying when tips are on.